File: estudos-pf/nivel-2-sequencias-e-comprehensions/solution/q2/q2.clj
;; [n.upper() for n in palavras if len(n) > 3]

(require '[clojure.string :as str])

(defn capitaliza-maior-que-3 [sequencia]
  (for [palavra sequencia :when (> (count palavra) 3)] (str/upper-case palavra)))

(defn map-filter-capitaliza-maior-que-3 [sequencia]
  (map str/upper-case (filter #(> (count %) 3) sequencia)))

(println (capitaliza-maior-que-3 '("oi" "fala")))
(println (map-filter-capitaliza-maior-que-3 '("oi" "fala")))

File: estudos-pf/nivel-2-sequencias-e-comprehensions/solution/q2/q2.py
palavras = ["banana", "maçã", "pera"]

list_comprehension = [n.upper() for n in palavras if len(n) > 3]

lista_map_filter = list(map(str.upper, filter(lambda n: len(n) > 3, palavras)))

assert list_comprehension == lista_map_filter
print(lista_map_filter)
print(list_comprehension)
